### PR TITLE
update volatile information

### LINKED ISSUE #5763: Example is irrelevent
I don't understand how the example shows the use or meaning of volatile. If I run it with or without volatile I'm getting the exact same results. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d226514d-c7cb-0d1b-b9dc-fb791399dc47
* Version Independent ID: aa5d1078-7f89-98ce-7fdc-107c9ed3eebc
* Content: [volatile (C# Reference)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile)
* Content Source: [docs/csharp/language-reference/keywords/volatile.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/keywords/volatile.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #5763

Relies on dotnet/samples#412

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening to force a new build (the samples PR is merged)

File: docs/csharp/language-reference/keywords/volatile.md [hunk 1]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
...
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Declaration)]
+
+The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
   
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Volatile)]
+
+With the `volatile` modifier on  the declaration of  in place, you'll always get the same results (similar to the excerpt shown in the preceding code). However, without that modifier on the `_shouldStop` member, the behavior is unpredictable. The `DoWork` method may optimize the member access, resulting in reading stale data. Because of the nature of multi-threaded programming, the number of stale reads is unpredictable. Different runs of the program will produce somewhat different results.
```
Comment (Thraka) [inline docs/csharp/language-reference/keywords/volatile.md]: I think variable names were here but removed. I see double spaces in the first start of the sentence, two of them.

**Review by Thraka — CHANGES_REQUESTED** [review]

File: docs/csharp/language-reference/keywords/volatile.md [hunk 2]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
...
-## Example  
- The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
+## Example
+
+The following example shows how to declare a public field variable as `volatile`.  
   
- [!code-csharp[csProgGuideThreading#1](../../../csharp/language-reference/keywords/codesnippet/CSharp/volatile_2.cs)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Declaration)]
+
+The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/volatile.md]: PR #8164 redirects the C# guide article to the .NET guide article. So the end of the sentence ` and [Threading (C#)](../../programming-guide/concepts/threading/index.md)` can be removed.

File: docs/csharp/language-reference/keywords/volatile.md [hunk 3]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
 ms.assetid: 78089bc7-7b38-4cfd-9e49-87ac036af009
 ---
 # volatile (C# Reference)
+
 The `volatile` keyword indicates that a field might be modified by multiple threads that are executing at the same time. Fields that are declared `volatile` are not subject to compiler optimizations that assume access by a single thread. These restrictions ensure that all threads will observe volatile writes performed by any other thread in the order in which they were performed. There is no guarantee of a single total ordering of volatile writes as seen from all threads of execution.  
   
- The `volatile` modifier is usually used for a field that is accessed by multiple threads without using the [lock](../../../csharp/language-reference/keywords/lock-statement.md) statement to serialize access.  
+ The `volatile` modifier is usually used for a field that is accessed by multiple threads without using the [lock](lock-statement.md) statement to serialize access.
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/volatile.md]: Is it worth to mention the [Interlocked](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked?view=netframework-4.7.2) class here?

File: docs/csharp/language-reference/keywords/volatile.md [hunk 4]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
...
--   Reference types.  
-  
--   Pointer types (in an unsafe context). Note that although the pointer itself can be volatile, the object that it points to cannot. In other words, you cannot declare a "pointer to volatile."  
+- Reference types.  
   
--   Types such as sbyte, byte, short, ushort, int, uint, char, float, and bool.  
+- Pointer types (in an unsafe context). Note that although the pointer itself can be volatile, the object that it points to cannot. In other words, you cannot declare a "pointer to volatile."  
   
--   An enum type with one of the following base types: byte, sbyte, short, ushort, int, or uint.  
+- Simple types such as `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `char`, `float`, and `bool`.
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/volatile.md]: Maybe here, it's worth to mention `Interlocked` for the `long` and `double` types as those types are supported by `volatile`

File: docs/csharp/language-reference/keywords/volatile.md [hunk 5]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
...
 ## See Also
 
-- [C# Reference](../../../csharp/language-reference/index.md)  
-- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
-- [C# Keywords](../../../csharp/language-reference/keywords/index.md)  
-- [Modifiers](../../../csharp/language-reference/keywords/modifiers.md)
+- [C# Reference](../index.md)  
+- [C# Programming Guide](../../programming-guide/index.md)  
+- [C# Keywords](index.md)  
+- [Modifiers](modifiers.md)
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/volatile.md]: Links to the `lock` keyword and xref to the `Interlocked` would be useful here, for sure.

And maybe that blog post, as well:
https://blogs.msdn.microsoft.com/ericlippert/2011/06/16/atomicity-volatility-and-immutability-are-different-part-three/

File: docs/csharp/language-reference/keywords/volatile.md [hunk 6]
```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
...
   
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Volatile)]
+
+With the `volatile` modifier on  the declaration of  in place, you'll always get the same results (similar to the excerpt shown in the preceding code). However, without that modifier on the `_shouldStop` member, the behavior is unpredictable. The `DoWork` method may optimize the member access, resulting in reading stale data. Because of the nature of multi-threaded programming, the number of stale reads is unpredictable. Different runs of the program will produce somewhat different results.
+
+## C# Language Specification
+
+[!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/volatile.md]: To follow #7890, please consider to give the following link:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#volatile-fields

File: docs/csharp/language-reference/keywords/volatile.md (hunk 3 above)
Comment (pkulikov) [reply docs/csharp/language-reference/keywords/volatile.md]: @BillWagner a couple more comments.

"...to serialize access." or "...to synchronize access."?

In general, this paragraph invites to use `volatile` when one needs to synchronize access to a shared field. It sounds like a risky trip. Maybe, this paragraph should be removed at all?

Comment (BillWagner) [reply docs/csharp/language-reference/keywords/volatile.md]: I did remove the paragraph. I also added more text from the C# spec in the introductory paragraph.

**Comment by BillWagner** [comment]
Updated per feedback. It's ready for another look @Thraka

File: docs/csharp/language-reference/keywords/volatile.md [hunk 7]
```diff
@@ -9,42 +9,44 @@ helpviewer_keywords:
...
   
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Volatile)]
+
+With the `volatile` modifier added t the declaration of `_shouldStop` in place, you'll always get the same results (similar to the excerpt shown in the preceding code). However, without that modifier on the `_shouldStop` member, the behavior is unpredictable. The `DoWork` method may optimize the member access, resulting in reading stale data. Because of the nature of multi-threaded programming, the number of stale reads is unpredictable. Different runs of the program will produce somewhat different results.
+
+## C# Language Specification
+
+[!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]
```
Comment (Thraka) [inline docs/csharp/language-reference/keywords/volatile.md]: Is this correct?

> added t the declaration?

Comment (BillWagner) [reply docs/csharp/language-reference/keywords/volatile.md]: dumb typo. Fixed.

**Comment by Thraka** [comment]
LGTM!

**Review by Thraka — APPROVED** [review]

**Pull request merged**